staging/istio/charts/security/templates/cert-delete-job.yaml

```diff
@@ -0,0 +1,51 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cert-delete-job
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: cert-delete-job
+  namespace: {{ .Release.Namespace }}
+rules:
+- apiGroups: ["certmanager.k8s.io"]
+  resources: ["certificates"]
+  verbs: ["get", "list", "watch", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: cert-delete-job
+  namespace: {{ .Release.Namespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: cert-delete-job
+subjects:
+  - kind: ServiceAccount
+    name: cert-delete-job
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: cert-delete-{{ randAlphaNum 5 | lower }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded,before-hook-creation
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false" 
+    spec:
+      serviceAccountName: cert-delete-job
+      containers:
+      - name: istio-ca-delete
+        image: docker.io/bitnami/kubectl:1.16.2
+        command: ["kubectl",  "delete", "certificate", "istio-ca", "-n", "{{ .Release.Namespace }}"]
```

Comment:
Are there situations where this certificate would NOT be present and would block deletion due to this command failing? Should we add a check here to succeed when the cert is not present, regardless of whether this job had to delete it?

Reply:
Doesn't hurt to add. So just added. PTAL